src/field_theory/minimal_polynomial.lean

```diff
@@ -90,6 +97,35 @@ begin
   simpa only [eq_one, alg_hom.map_one, one_ne_zero] using aeval hx
 end
 
+/--If L/K is a ring extension, and x is an element of L in the image of K,
+then the minimal polynomial of x is X - C x.-/
+lemma algebra_map_inj [nontrivial α] (a : α) (hf : function.injective (algebra_map α β))
+  (ha : is_integral α (algebra_map α β a)) : minimal_polynomial ha = X - C a :=
+begin
+  have hdegle : (minimal_polynomial ha).nat_degree ≤ 1,
+  { apply with_bot.coe_le_coe.1,
+    rw [←degree_eq_nat_degree (ne_zero ha), with_top.coe_one],
+    rw [←degree_X_sub_C a],
```

Comment:
Should these be combined or are they acting on different goals? (In the latter case, they should be surrounded by braces).